src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -508,12 +515,7 @@ PostProcessors getPostProcessors() {
 
   public WriteModelStrategy getWriteModelStrategy() {
```

Comment:
Seems strange to have two methods with the same name. But having different effects (one caching). I'd be tempted to refactor / rename eg: `getWriteModelStrategyFromConfig`